src/emulator/gui/src/settings_dialog.cpp

```diff
@@ -48,13 +49,83 @@ bool change_pref_location(const std::string &input_path, const std::string &curr
     return true;
 }
 
+std::vector<std::string>::iterator modules;
+vfs::FileBuffer buffer;
+static ImGuiTextFilter search_bar;
```

Comment:
Bold reason to put this global. Please avoid doing this when you can.
Try to put these variables to function